src/twig/linearize/Callable/test/Callable.test.cpp

```diff
@@ -28,33 +37,20 @@ SCENARIO("Broken stick implementation"){
     return 0.5 * ( std::get<0>(x) + std::get<1>(x) );
   };
 
-  auto linearization = linearize::callable< double >( instance );
+  auto linearization = linearize::callable( xInstance, yInstance );
   auto first = xgrid.begin();
   auto last = xgrid.end();
   linearization( first, last, functor, criterion, midpoint );
   CHECK( first == last );
 
   // Make sure all the original x values are in the linearized x
   for( double& x : xgrid ){
-    CHECK( std::any_of( instance.begin(), instance.end(), 
+    CHECK( std::any_of( xInstance.begin(), xInstance.end(), 
                         [&]( auto& arg ){ return arg == x; } ) );
   }
   
-  auto iterator = instance.begin();
-
-  auto left = [&iterator](){ return iterator[0]; };
-  auto right = [&iterator](){ return iterator[1]; };
-
-  auto x = []( auto value ){ return value; };
-  auto y = [&]( auto value ){ return functor( value ); };
-
-  while ( iterator != std::prev( instance.end() ) ){
-    const auto midpoint = 0.5 * ( x( left() ) + x( right() ) );
-    const auto trial = 0.5 * ( y( left() ) + y( right() ) );
-    const auto reference = y( midpoint );
-    CHECK( criterion( trial, reference,
-                       x( left() ), x( right() ),
-                       y( left() ), y( right() ) ) );
-    ++iterator;
+  auto length = xInstance.size();
```

Comment:
You should still check that the vectors are the same size after the linearisation. The for loop that follows assumes that there are no additional elements in the y vector but that is not the same as verifying that they are the same length.

Reply:
I've added some tests on this.